models/transfer.py

```
@@ -113,39 +127,181 @@ def get(self, transfer_id: str) -> Optional[Dict]:
+            return 0, []
+
+        deleted, skipped = 0, []
+        with self.db.get_connection() as conn:
+            for start in range(0, len(transfer_ids), DELETE_BATCH):
+                batch = transfer_ids[start:start + DELETE_BATCH]
+                placeholders = ', '.join('?' for _ in batch)
+
+                running = [
+                    row['transfer_id'] for row in conn.execute(
+                        f"SELECT transfer_id FROM transfers"
+                        f" WHERE transfer_id IN ({placeholders}) AND status = 'running'", batch
```

Ruff (0.16.0)

[error] Possible SQL injection vector through string-based query construction

(S608)